src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,156 +27,159 @@ case class CromwellRuntimeAttributes(docker: Option[String],
 object CromwellRuntimeAttributes {
   private val log = LoggerFactory.getLogger("RuntimeAttributes")
 
-  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): CromwellRuntimeAttributes = {
-    val attributeMap = AttributeMap(wdlRuntimeAttributes.attrs)
+  def apply(wdlRuntimeAttributes: RuntimeAttributes, backendCall: BackendCall): Try[CromwellRuntimeAttributes] = {
+    for {
+      keys <- validateKeys(wdlRuntimeAttributes, backendCall.backend.backendType)
+      attributes <- TryUtil.sequenceMap(wdlRuntimeAttributes.evaluate(backendCall.lookupFunction(backendCall.locallyQualifiedInputs), backendCall.engineFunctions))
+      validatedAttributes <- validateRuntimeAttributes(attributes)
+    } yield validatedAttributes
+  }
+
+  def validateKeys(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): Try[RuntimeAttributes] = {
```

Comment:
I didn't really notice that these were public (but I still say public library API should throw). 

Appy should definitely be throwing, however